zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/repo/AzureNotebookRepo.java

```diff
@@ -0,0 +1,201 @@
+ * limitations under the License.
+ */
+
+package org.apache.zeppelin.notebook.repo;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.microsoft.azure.storage.CloudStorageAccount;
+import com.microsoft.azure.storage.StorageException;
+import com.microsoft.azure.storage.file.*;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.zeppelin.conf.ZeppelinConfiguration;
+import org.apache.zeppelin.notebook.Note;
+import org.apache.zeppelin.notebook.NoteInfo;
+import org.apache.zeppelin.notebook.Paragraph;
+import org.apache.zeppelin.scheduler.Job;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.io.*;
+import java.net.URISyntaxException;
+import java.security.InvalidKeyException;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Azure storage backend for notebooks
+ */
+public class AzureNotebookRepo implements NotebookRepo {
+  private static final Logger LOG = LoggerFactory.getLogger(S3NotebookRepo.class);
+
+  private final ZeppelinConfiguration conf;
+  private final String user;
+  private final String shareName;
+  private final CloudFileDirectory rootDir;
+
+  public AzureNotebookRepo(ZeppelinConfiguration conf)
+      throws URISyntaxException, InvalidKeyException, StorageException {
+    this.conf = conf;
+    user = conf.getString(ZeppelinConfiguration.ConfVars.ZEPPELIN_NOTEOOK_AZURE_USER);
+    shareName = conf.getString(ZeppelinConfiguration.ConfVars.ZEPPELIN_NOTEOOK_AZURE_SHARE);
+
+    CloudStorageAccount account = CloudStorageAccount.parse(
+        conf.getString(ZeppelinConfiguration.ConfVars.ZEPPELIN_NOTEBOOK_AZURE_CONNECTION_STRING));
+    CloudFileClient client = account.createCloudFileClient();
+    CloudFileShare share = client.getShareReference(shareName);
+    share.createIfNotExists();
+
+    CloudFileDirectory userDir = StringUtils.isBlank(user) ?
+        share.getRootDirectoryReference() :
+        share.getRootDirectoryReference().getDirectoryReference(user);
+    userDir.createIfNotExists();
+
+    rootDir = userDir.getDirectoryReference("notebook");
+    rootDir.createIfNotExists();
+  }
+
+  @Override
+  public List<NoteInfo> list() throws IOException {
+    List<NoteInfo> infos = new LinkedList<NoteInfo>();
+    NoteInfo info = null;
+
+    for (ListFileItem item : rootDir.listFilesAndDirectories()) {
+      if (item.getClass() == CloudFileDirectory.class) {
+        CloudFileDirectory dir = (CloudFileDirectory) item;
+
+        try {
+          if (dir.getFileReference("note.json").exists()) {
+            info = new NoteInfo(getNote(dir.getName()));
+
+            if (info != null) {
+              infos.add(info);
+            }
+          }
+        } catch (URISyntaxException e) {
+          LOG.error("Error enumerating notebooks", e);
+        } catch (StorageException e) {
+          LOG.error("Error enumerating notebooks", e);
```

Comment:
Could you put different error log text or combine the catch?
`catch (URISyntaxException | StorageException e) {`
https://docs.oracle.com/javase/tutorial/essential/exceptions/catch.html